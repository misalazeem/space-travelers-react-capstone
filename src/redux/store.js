import { configureStore } from '@reduxjs/toolkit';
import missionSlice, { loadMissions } from './missions/missionSlice';

export const store = configureStore({
  reducer: {
    missions: missionSlice,
  },
});

const mission = { mission_id: '1', mission_name: 'Apollo 13', description: 'Mission to the Moon' };
store.dispatch(loadMissions(mission));

export default store;
