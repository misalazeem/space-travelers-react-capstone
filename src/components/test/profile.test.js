import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../redux/store';
import Profile from '../profile.jsx';

test('renders Profile component correctly', () => {
  const { container } = render(
    <Provider store={store}>
      <Profile />
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
