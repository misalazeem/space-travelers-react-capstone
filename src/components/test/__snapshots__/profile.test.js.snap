// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Profile component correctly 1`] = `
<div>
  <div
    class="profile-section"
  >
    <div
      class="profile-missions"
    >
      <h2>
        My Missions
      </h2>
      <p>
        No Missions Joined
      </p>
    </div>
    <div
      class="profile-rockets"
    >
      <h2>
        My Rockets
      </h2>
      <p>
        No Rockets Reserved
      </p>
    </div>
  </div>
</div>
`;
